### DISCUSSION TITLE
How to handle models with no ID?

### DISCUSSION THREAD

**Discussion by rdk8** [joshhanley/livewire-autocomplete#14] [Help]
Hi Josh,

I haven't dug into the code yet, but I don't see anything relevant in the documentation. Is it possible to handle models that don't have an ID field? In my case I am using a view that returns a list of names without IDs. If I omit the ID prop, I get an undefined column error where it's looking for an ID column.

Thanks,
Raja

**Comment by joshhanley** [comment]
@rdk8 hmm I think you would still need some way to distinguish one model from the next. So if your list of names are unique, you could use `name` as the identifier as well as using name for the display.

To do that, pass an array into the options parameter with a key of `id` value of `name` (or whatever the property on your model will be unique)

```blade
<x-lwa::autocomplete
...
:options="[
        'id' => 'name',
]"
```

That would then essentially work similar to if you had a standard html select with the option value and label the same, like `<option value="{{ $model->name }}">{{ $model->name }}</option>`

Hope this helps!
